tests/src/test/scala/com/nvidia/spark/rapids/RapidsShuffleIntegrationSuite.scala

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.funsuite.AnyFunSuite
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+
+/**
+ * Integration test suite for RAPIDS Shuffle Manager.
+ * Tests buffer fallback behavior under different configurations:
+ * - Verifies SpillablePartialFileHandle memory-to-disk fallback
+ * - Tests different partialFileBufferMaxSize settings
+ * - Validates forced file-only mode for finalMergeWriter
+ */
+class RapidsShuffleIntegrationSuite extends AnyFunSuite with BeforeAndAfterEach {
+
+  private var spark: SparkSession = _
+  
+  /**
+   * Capture log messages during operation and check for buffer behavior indicators.
+   * 
+   * @param operation The operation to execute while capturing logs
+   * @return Tuple of (hasExpansion, hasSpill, hasForcedFileOnly)
+   *         - hasExpansion: Whether buffer expansion occurred
+   *         - hasSpill: Whether buffer spilled to disk
+   *         - hasForcedFileOnly: Whether forced file-only mode was used
+   */
+  private def checkLogsForBufferBehavior(operation: => Unit): 
+      (Boolean, Boolean, Boolean) = {
+    val loggerNames = Seq(
+      "com.nvidia.spark.rapids.spill.SpillablePartialFileHandle",
+      "org.apache.spark.shuffle.sort.io.RapidsLocalDiskShuffleMapOutputWriter"
+    )
+    
+    val logMessages = LogCaptureUtils.captureLogsFrom(loggerNames) {
+      operation
+    }
+    
+    val hasExpansion = logMessages.exists(_.contains("Expanded buffer from"))
```

Comment:
I am not a big fan of using log messages in a test like this. For me this is a really brittle way of validating that code is working because I would never expect that if I changed a log message that a test would fail. I would much rather use metrics or some abstract methods that we can mock/override to get the visibility we need/want.